doc/source/whatsnew/v0.23.0.txt

```diff
@@ -232,6 +232,81 @@ Build Changes
 - Building from source now explicitly requires ``setuptools`` in ``setup.py`` (:issue:`18113`)
 - Updated conda recipe to be in compliance with conda-build 3.0+ (:issue:`18002`)
 
+Extraction of matching patterns from strings
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, extracting matching patterns from strings with :func:`str.extract` used to return a
+``Series`` if a single group was being extracted (a ``DataFrame`` if more than one group was
+extracted``). As of Pandas 0.23.0 :func:`str.extract` always returns a ``DataFrame``, unless
+``expand`` is set to ``False`` (:issue:`11386`).
+
+Also, ``None`` was an accepted value for the ``expand`` parameter (which was equivalent to
+``False``), but now raises a ``ValueError``.
+
+Previous Behavior:
+
+.. code-block:: ipython
+
+    In [1]: s = pd.Series(['number 10', '12 eggs'])
+
+    In [2]: extracted = s.str.extract('.*(\d\d).*')
+
+    In [3]: extracted
+    Out [3]:
+    0    10
+    1    12
+    dtype: object
+
+    In [4]: type(extracted)
+    Out [4]:
+    pandas.core.series.Series
+
+    In [5]: s.str.extract('.*(\d\d).*', expand=None)
+    Out [5]:
+    0    10
+    1    12
+    dtype: object
+
+New Behavior:
+
+.. code-block:: ipython
```

Comment:
Can you make this a `.. ipython:: python` directive? (and remove the output below) ?

Reply:
Should I use the `..ipython:: python` directive in all the 3 cases, right?

I'm not sure what do you mean by "remove the output below", sorry. Can you clarify please?

Reply:
To clarify: you can use it for code that can run *now*, so for the new behaviour. For showing the "old behaviour" you still need to use the literal code-block (there you don't need to change anything). 

And by removing output I mean that in an `ipython:: python` block, you only include the actual code, as the output of the code is generated during the doc build. If you scroll a bit up in the file, you see some examples of that.